### ISSUE TITLE
Shrink domain object code using Immutables

### ISSUE THREAD

**Issue by whitfin** [lake54/groupsio-api-java#5]
Currently the domain code is huge, due to builder methods, getters, setters, etc. This makes adding new domain stuff kinda painful.

I think that it would be nice if we were to make use of https://immutables.github.io/ to shrink down the code per model and get a lot of that stuff for free. 

Unless you object, I plan on PR'ing this as soon as I get free time.

**Comment by lake54** [comment]
Agreed for the most part - they were generated with jsonschema2pojo.org so initially creating them was 'trivial', but I can see how adding new attributes wouldn't be as straightforward. Happy to review some changes around this, but not sure how far I'd be willing to go from the current implementation(s) for the sake of LOC.
